src/components/ResourcePage.tsx

```diff
@@ -82,9 +82,10 @@ const EditResourceRedirect = <T extends BaseResource>({
   const { pathname } = useLocation();
   const locale = i18n.language;
   const { id } = useParams<'id'>();
-  const { data, error, isLoading } = useHook(id!, undefined, { enabled: !!id });
+  const parsedId = Number(id);
+  const { data, error, isLoading } = useHook(parsedId, undefined, { enabled: !!parsedId });
```

Comment:
Her ville jeg heller hatt to linjer: 
`const params = useParams<'id'>();`
`const id = Number(params.id) || undefined`